components/StreamField/types.ts

```diff
@@ -3,5 +3,5 @@ import type { Topic } from './blocks/TopicsBlock';
 
 export type StreamFieldProps = {
     body: Array<StreamFieldBlock>;
-    topics: Array<Topic>;
+    topics?: Array<Topic>;
```

Comment:
**Question**

Now I'm just wondering - shouldn't we fetch `topics` in `TopicsBlock` instead of passing them to a StreamField?

Reply:
Do you mean using something like `useEffect` as suggested [here](https://nextjs.org/docs/basic-features/data-fetching/client-side#client-side-data-fetching-with-useeffect) in next.js docs?

In the link above, we can read that:
> Client-side data fetching is useful when your page doesn't require SEO indexing, when you don't need to pre-render your data, or when the content of your pages needs to update frequently.

And also that:
> It's important to note that using client-side data fetching can affect the performance of your application and the load speed of your pages. This is because the data fetching is done at the time of the component or pages mount, and the data is not cached.

I think we want to use server-side data fetching considering all this.

Reply:
Sounds good!